engine/src/main/java/nl/inl/blacklab/forwardindex/TermsReader.java

```diff
@@ -210,15 +215,21 @@ private void fillTermCharData(String[] terms) {
         byte[][] termCharData = new byte[0][];
         byte[] curArray;
         for (int termIndex = 0; termIndex < numberOfTerms; ++termIndex) {
+
             // allocate new term bytes array, subtract what will fit
             final int curArrayLength = (int) Long.min(bytesRemainingToBeWritten, Integer.MAX_VALUE);
             curArray = new byte[curArrayLength];
 
             // now write terms until the array runs out of space or we have written all remaining terms
+            // FIXME this code breaks when char term data total more than 2 GB
+            //       (because offset will overflow)
```

Comment:
I am probably missing something here, but this will overflow, after the first iteration no?

Reply:
Yes, I noticed that too and added this comment and created #282 because the fix wasn't immediately obvious to me. Right now it is not an issue unless our largest index grows by another factor of 4, so you should never run into it I think. We'll look into this and fix it soon.